pyiceberg/catalog/__init__.py

```diff
@@ -587,8 +590,34 @@ def _write_metadata(metadata: TableMetadata, io: FileIO, metadata_path: str) ->
         ToOutputFile.table_metadata(metadata, io.new_output(metadata_path))
 
     @staticmethod
-    def _get_metadata_location(location: str) -> str:
-        return f"{location}/metadata/00000-{uuid.uuid4()}.metadata.json"
+    def _get_metadata_location(location: str, new_version: int = 0) -> str:
+        if new_version < 0:
+            raise ValueError(f"Table metadata version: {new_version} cannot be negative")
```

Comment:
Should we also return a more friendly error in the off chance that new_version is `None`?
```suggestion
        if not new_version or new_version < 0:
            raise ValueError(f"Table metadata version: {new_version} must be a positive integer")
```